prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```
@@ -370,17 +372,14 @@
                 )
                 self.save_answer_to_long_term_memory(prediction)
 
-        final_answer = (
-            self.generate_final_decision(
-                question, scenarios_with_probs, created_time=created_time
-            )
-            if scenarios_with_probs
-            else None
+            if len(scenarios_with_probs) < len(all_scenarios) / 2:
+                raise ValueError(
+                    f"Too many of sub_predictions have failed, stopping the agent."
```

Ruff

f-string without any placeholders

Remove extraneous `f` prefix

(F541)